TLM/TLM/Traffic/Impl/SegmentEnd.cs

```diff
@@ -82,14 +82,14 @@ public SegmentEnd(ushort segmentId, bool startNode)
             // TODO pre-calculate this
             uint avgSegLen = (uint)SegmentId.ToSegment().m_averageLength;
 
-            IDictionary<ushort, uint>[] ret =
+            Dictionary<ushort, uint>[] ret =
                 includeStopped ? numVehiclesGoingToSegmentId : numVehiclesMovingToSegmentId;
 
             // reset
             for (byte laneIndex = 0; laneIndex < ret.Length; ++laneIndex) {
                 IDictionary<ushort, uint> laneMetrics = ret[laneIndex];
-                foreach (KeyValuePair<ushort, uint> e in laneMetrics) {
-                    laneMetrics[e.Key] = 0;
+                foreach (ushort key in laneMetrics.Keys.ToList()) {
```

Comment:
eh what?
you don't have to copy the keys to iterate over them do you?
you can use the keyvaluepair too.
I think its a struct.